gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```diff
@@ -0,0 +1,398 @@
+
+    // Users can specify the BigLake catalog ID, otherwise catalog plugin will be used.
+    this.catalogId = properties.getOrDefault(PROPERTIES_KEY_BLMS_CATALOG, inputName);
+    this.catalogName = CatalogName.of(projectId, location, catalogId);
+    LOG.info("Use BigLake catalog: {}", catalogName.toString());
+
+    String fileIOImpl =
+        properties.getOrDefault(CatalogProperties.FILE_IO_IMPL, ResolvingFileIO.class.getName());
+    this.fileIO = CatalogUtil.loadFileIO(fileIOImpl, properties, conf);
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier identifier) {
+    // The identifier of metadata tables is like "ns.table.files".
+    // We return a non-existing table in this case (empty table ID is disallowed in BigLake
+    // Metastore), loadTable will try loadMetadataTable.
+    if (identifier.namespace().levels().length > 1
+        && MetadataTableType.from(identifier.name()) != null) {
+      return new BigLakeTableOperations(
+          client, fileIO, getTableName(identifier.namespace().level(0), /* tableId= */ ""));
+    }
+
+    return new BigLakeTableOperations(
+        client,
+        fileIO,
+        getTableName(getDatabaseId(identifier.namespace()), /* tableId= */ identifier.name()));
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier identifier) {
+    String locationUri = getDatabase(identifier.namespace()).getHiveOptions().getLocationUri();
+    return String.format(
+        "%s/%s",
+        Strings.isNullOrEmpty(locationUri)
+            ? getDatabaseLocation(getDatabaseId(identifier.namespace()))
+            : locationUri,
+        identifier.name());
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    // When deleting a BLMS catalog via `DROP NAMESPACE <catalog>`, this method is called for
+    // verifying catalog emptiness. `namespace` is empty in this case, we list databases in
+    // this catalog instead.
+    // TODO: to return all tables in all databases in a BLMS catalog instead of a "placeholder".
+    if (namespace.levels().length == 0) {
+      return Iterables.isEmpty(client.listDatabases(catalogName))
+          ? ImmutableList.of()
+          : ImmutableList.of(TableIdentifier.of("placeholder"));
+    }
+
+    return Streams.stream(client.listTables(getDatabaseName(namespace)))
+        .map(BigLakeCatalog::getTableIdentifier)
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    TableOperations ops = newTableOps(identifier);
+    // TODO: to catch NotFoundException as in https://github.com/apache/iceberg/pull/5510.
+    TableMetadata lastMetadata = ops.current();
+    try {
+      client.deleteTable(
+          getTableName(getDatabaseId(identifier.namespace()), /* tableId= */ identifier.name()));
+    } catch (NoSuchTableException e) {
+      LOG.warn("Dropping table failed", e);
```

Comment:
This isn't a failure, it's supported by the API. Can you remove this log message?

Reply:
Done.